wordpress-mobile/gutenberg-mobile
# Detect content changed

Addresses part of #182 

Fit the Bridge interface with a extra boolean to denote if the returned html is changed compared to the original html the editor was started with. Parent app can then optimize for not the case where the content didn't change (e.g. not marking the post as changed, save it and so on).

### Changes in this PR

* State adopted the `initialHtmlHash` member. Reducer code adapted to support it.
* Html change detection is using `md5` hashes to compare the html versions
* `BlockType` and `StateType` moved to own file. Similar with a couple of state util functions.
* Bridge middleware compares html hashes on BLOCK.SERIALIZE_ALL, to report to parent app

### To test A:

App should compile and run as usual. No visible feature change.

### To test B (Android only):

* Use the `gb/detect-content-change` branch from wpandroid. Usual `submodule update`, `yarn install`, `yarn start:reset` dance to launch Metro.
* `yarn wpandroid` to launch wpandroid
* Open a Gutenberg post that has no local changes. Don't make any changes and go "Back".
* Notice the post retaining the status it had before, without marking the post as having local changes.

cc @etoledom , @pinarol as this includes Bridge changes to pass the "changed" flag.
